unflake TestIntegration_UpdateRetentionPolicy

https://sponge.corp.google.com/target?id=a8d868bf-56bf-4ede-972f-48bbf660a9ba&target=cloud-go-libraries/google-cloud-go/continuous/go112&searchFor=&show=ALL&sortBy=STATUS

```
=== RUN   TestIntegration_UpdateRetentionPolicy
--- FAIL: TestIntegration_UpdateRetentionPolicy (19.96s)
    integration_test.go:2974: integration_test.go:2508: update: googleapi: Error 429: The project exceeded the rate limit for creating and deleting buckets., rateLimitExceeded
```

Fix didn't work correctly in Kokoro:
```
=== RUN   TestIntegration_UpdateRetentionPolicy
    TestIntegration_UpdateRetentionPolicy: integration_test.go:2641: input: <nil>
        got=-, want=+:
          (*storage.RetentionPolicy)(
        - 	nil,
        + 	&{RetentionPeriod: s"1m0s"},
          )
--- FAIL: TestIntegration_UpdateRetentionPolicy (7.98s)
```

Reverted and re-opening this issue.

--- a/storage/integration_test.go
+++ b/storage/integration_test.go
@@ -2644,7 +2644,6 @@ func TestIntegration_UpdateRetentionExpirationTime(t *testing.T) {
 }
 
 func TestIntegration_UpdateRetentionPolicy(t *testing.T) {
-	t.Skip("https://github.com/googleapis/google-cloud-go/issues/1632")
 	ctx := context.Background()
 	client := testConfig(ctx, t)
 	defer client.Close()

